engine/src/main/java/org/terasology/engine/core/module/ModuleManager.java

```diff
@@ -136,6 +146,35 @@ private void loadModulesFromApplicationPath(PathManager pathManager) {
         scanner.scan(registry, paths);
     }
 
+    private void loadModulesFromClassPath() {
+        ClasspathCompromisingModuleFactory moduleFactory = (ClasspathCompromisingModuleFactory) this.moduleFactory;
+        for (String metadata_name : moduleFactory.getModuleMetadataLoaderMap().keySet()) {
+            Enumeration<URL> urls;
+            try {
+                urls = ClassLoader.getSystemResources(metadata_name);
+            } catch (IOException e) {
+                throw new RuntimeException(e);
```

Comment:
What's the reason we turn the `IOException` into a `RuntimeException` here? Usually a runtime exception is something that could have been prevented by the caller - is that the case here?

Or is it just to have an unchecked exception that bubbles up without annotating `throws ...` everywhere?

We should probably start building a common set of exceptions we want to use in the code base, with dedicated error handling and meaningful logging....

Reply:
Hm, maybe just using `RuntimeException` is actually good enough because the _cause_ is preserved when handing the `e` through :thinking: 

https://softwareengineering.stackexchange.com/questions/121328/is-it-good-practice-to-catch-a-checked-exception-and-throw-a-runtimeexception

Reply:
> Or is it just to have an unchecked exception that bubbles up without annotating throws ... everywhere?

Mostly this.

I've been pretty sloppy about things like this, figuring that we can defer the decision to make well-typed exceptions until we see which ones come up in practice and we have ideas about how to deal with them.

One consideration is how likely they are. An IO error as a result of a file operation to a user-supplied location is one thing. But a read-only `getSystemResources` call? Seems like the environment would have to be pretty broken for that to happen. Doesn't make much sense to me to document that as part of this method's interface.